File: apps/web/src/features/organizations/types/organization.ts
/**
 * Organization types and interfaces
 */

export type OrganizationStatus = "active" | "suspended";
export type PlanTier = "free" | "pro" | "enterprise";

export interface Organization {
  id: string;
  name: string;
  slug: string;
  status: OrganizationStatus;
  plan_tier: PlanTier;
  created_by: string;
  created_at: string;
  updated_at: string;
}

export interface OrganizationWithStats extends Organization {
  user_count?: number;
  agent_count?: number;
}

export interface CreateOrganizationInput {
  name: string;
  slug: string;
  plan_tier?: PlanTier;
}

export interface UpdateOrganizationInput {
  name?: string;
  slug?: string;
  plan_tier?: PlanTier;
  status?: OrganizationStatus;
}

export interface OrganizationFilters {
  status?: OrganizationStatus;
  search?: string;
  limit?: number;
  offset?: number;
}

export interface OrganizationResponse {
  records: Organization[];
  total: number;
}

File: src/studio/models/organization.py
"""
Organization DataFlow Model

Represents an organization/tenant in the Kaizen Studio platform.
"""

from studio.models import db


@db.model
class Organization:
    """
    Organization model for multi-tenant support.

    SSO domain grouping:
    - sso_domain: Primary SSO domain associated with this organization
    - allow_domain_join: Whether users from verified domains can auto-join
    - Use OrganizationDomain table for multiple verified domains

    DataFlow generates these nodes automatically:
    - OrganizationCreateNode
    - OrganizationReadNode
    - OrganizationUpdateNode
    - OrganizationDeleteNode
    - OrganizationListNode
    - OrganizationCountNode
    - OrganizationUpsertNode
    - OrganizationBulkCreateNode
    - OrganizationBulkUpdateNode
    - OrganizationBulkDeleteNode
    - OrganizationBulkUpsertNode
    """

    # Primary key - MUST be 'id' for DataFlow
    id: str

    # Organization details
    name: str
    slug: str

    # Status: active, suspended, deleted
    status: str

    # Plan tier: free, pro, enterprise
    plan_tier: str

    # Creator reference
    created_by: str

    # Timestamps (ISO 8601 strings for DataFlow)
    created_at: str
    updated_at: str

    # SSO domain grouping fields
    # Primary SSO domain (e.g., "company.com")
    sso_domain: str | None = None

    # Whether users from verified domains can auto-join
    allow_domain_join: bool = False

    # Organization settings (JSON string for custom configuration)
    settings: str | None = None
